chain/chain.go

```diff
@@ -38,26 +38,47 @@ type RPCClient struct {
 	quitMtx sync.Mutex
 }
 
-// NewRPCClient creates a client connection to the server described by the
+// NewRPCClient creates a direct client connection to the server described by the
 // connect string.  If disableTLS is false, the remote RPC certificate must be
 // provided in the certs slice.  The connection is not established immediately,
 // but must be done using the Start method.  If the remote server does not
 // operate on the same decred network as described by the passed chain
 // parameters, the connection will be disconnected.
 func NewRPCClient(chainParams *chaincfg.Params, connect, user, pass string, certs []byte,
 	disableTLS bool) (*RPCClient, error) {
+	return NewClient(chainParams, connect, user, pass,
+		certs, disableTLS, "", "", "")
+}
+
+// NewClient creates a client connection to the server described by the
+// connect string.  If disableTLS is false, the remote RPC certificate must be
+// provided in the certs slice.  The connection is not established immediately,
+// but must be done using the Start method.  If the remote server does not
+// operate on the same decred network as described by the passed chain
+// parameters, the connection will be disconnected. If proxypath is not empty,
+// it will be used to connect.
+func NewClient(chainParams *chaincfg.Params, connect, user, pass string, certs []byte,
+	disableTLS bool, proxypath, proxyuser, proxypass string) (*RPCClient, error) {
+	return NewClientConfig(chainParams, &dcrrpcclient.ConnConfig{
+		Host:                 connect,
+		Endpoint:             "ws",
+		User:                 user,
+		Pass:                 pass,
+		Certificates:         certs,
+		DisableAutoReconnect: true,
+		DisableConnectOnNew:  true,
+		DisableTLS:           disableTLS,
+		Proxy:                proxypath,
+		ProxyUser:            proxyuser,
+		ProxyPass:            proxypass,
+	})
+}
 
+// NewClientConfig creates a client connection to the server described by the
+// passed chainParams and connConfig
+func NewClientConfig(chainParams *chaincfg.Params, connConfig *dcrrpcclient.ConnConfig) (*RPCClient, error) {
```

Comment:
nitpick: can you rename this to NewRPCClientConfig for more similarity with the existing NewRPCClient?